components/eam/src/physics/yaml/nucleate_ice_cam/f90_yaml/nucleate_ice_cam_calc_beg_yml.f90

```diff
@@ -0,0 +1,101 @@
+  yaml%nstep_print = 1402 !time step(**remove these if generating data for a dependent subroutines**)
+
+  ! set ext_str if there are multiple sets of yaml output to be write out
+  ! here gives an example that "flag" in the code can be 0, 1, or 2:
+  ! write(ext_str,'(I2)') flag
+  ! ext_str = 'flag_'//adjustl(ext_str)
+
+
+  !YAML file input generation code- DO NOT PORT to C++
+  !print all inputs one-by-one at column "yaml%col_print"
+  yaml%col_print = icolprnt(lchnk) !column to write data(**remove these if generating data for a dependent subroutines**)
+  y_nstep = get_nstep() !time step (**remove these if generating data for a dependent subroutines**)
+
+  yaml%flag_print = .false. ! to write or not to write data (**remove these if generating data for a dependent subroutines**)
+  !if(yaml%col_print == y_i .and. y_nstep==yaml%nstep_print .and. y_k == yaml%lev_print) then ! if this column exists in y_lchnk
+     !<
+     !In the case of y_i or y_k are not passed as arguments:
+  if(yaml%col_print >0 .and. y_nstep==yaml%nstep_print) then
+
+     !For generating data for a dependent subroutines where "yaml" derived type is already initialized:
+     !if(yaml%flag_print) then
+     !>
+
+     ! record number of calls that can output yaml file. you only need to write one set of input/output
+     n_calls = n_calls+1
+     if (n_calls==1) then ! output the first call. change this if writing out other calls
+     ! if ((n_calls==1 .and. flag==0) .or. (n_calls==3 .and. flag==1) .or. (n_calls==5 .and. flag==2)) then ! this is an example of writing multiple sets of YAML files
+
+        yaml%lchnk_print = lchnk !(**remove these if generating data for a dependent subroutines**)
+        yaml%flag_print  = .true.!(**remove these if generating data for a dependent subroutines**)
+
+        !open I/O yaml files (it can have an extra optional argument to pass a unique string to differentiate file names)
+        call open_files('nucleate_ice_cam_calc', &  !intent-in
+             unit_input, unit_output) !intent-out
+!             unit_input, unit_output, trim(ext_str)) !intent-out, with the use of ext_str
+
+
+        !start by adding an input string
+        call write_input_output_header(unit_input, unit_output,yaml%lchnk_print,yaml%col_print, &
+             'compute_tendencies',yaml%nstep_print, yaml%lev_print)
+
+        !< add code for writing data here>
+        call write_var(unit_input, unit_output, 'modeptr_accum', modeptr_accum)
+        call write_var(unit_input, unit_output, 'modeptr_aitken', modeptr_aitken)
+        call write_var(unit_input, unit_output, 'modeptr_coarse', modeptr_coarse)
+        call write_1d_var(unit_input, unit_output, 'numptr_amode', ntot_amode, numptr_amode(:))         
+        call write_1d_var(unit_input, unit_output, 'alnsg_amode', ntot_amode, alnsg_amode(:))
+        
+        call write_1d_var(unit_input, unit_output, 'lptr_dust_a_amode', ntot_amode, lptr_dust_a_amode(:))
```

Comment:
Could you write "subgrid_in" for nucleate_ice_cam?

Reply:
Do you mean nulceate_ice_subgrid? subgrid_in or subgrid is in nucleate_ice.F90, and is the same as nucleate_ice_subgrid. I have output it (subgrid) which is used in subroutine nucleati and hf in nucleate_ice.F90. It is just a namelist input value. I don't think it is used in the calculation of nucleate_ice_cam_calc.

Reply:
nulceate_ice_subgrid (or subgrid) is used by nucleati that is called in nucleate_ice_cam_calc. Because all validation tests in mam4xx are independent, we cannot get inputs from another test. I could copy and paste subgrid from nucleati validation test. However, It would be easier to have nulceate_ice_subgrid as an input parameter in nucleate_ice_cam_calc yaml files.

Reply:
Okay. I see. Just added in the input.

Reply:
Thanks